extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
 import org.exist.xslt.TransformerFactoryAllocator;
 import org.w3c.dom.Document;
 import org.xml.sax.SAXException;
+import uk.ac.ic.doc.slurp.multilock.MultiLock;
 
 public class Sync extends BasicFunction {
 
-	public final static FunctionSignature signature =
-		new FunctionSignature(
-            new QName("sync", FileModule.NAMESPACE_URI, FileModule.PREFIX),
-            "Synchronize a collection with a directory hierarchy. Compares last modified time stamps. " +
-            "If $dateTime is given, only resources modified after this time stamp are taken into account. " +
-    		"This method is only available to the DBA role.",
-            new SequenceType[]{
-            	new FunctionParameterSequenceType("collection", Type.STRING, 
-                        Cardinality.EXACTLY_ONE, "The collection to sync."),
-                new FunctionParameterSequenceType("targetPath", Type.ITEM, 
-                        Cardinality.EXACTLY_ONE, "The full path or URI to the directory"),
-                new FunctionParameterSequenceType("dateTime", Type.DATE_TIME, 
-                        Cardinality.ZERO_OR_ONE,
-                		"Optional: only resources modified after the given date/time will be synchronized.")
-            },
-            new FunctionReturnSequenceType(Type.BOOLEAN, Cardinality.EXACTLY_ONE, "true if successful, false otherwise")
-        );
-	
-	private final static Properties DEFAULT_PROPERTIES = new Properties();
-	static {
-		DEFAULT_PROPERTIES.put(OutputKeys.INDENT, "yes");
-		DEFAULT_PROPERTIES.put(OutputKeys.OMIT_XML_DECLARATION, "no");
+    public static final String PRUNE_OPT = "prune";
+    public static final String AFTER_OPT = "after";
+    public static final String EXCLUDES_OPT = "excludes";
+
+    public static final QName FileSyncElement = new QName("sync", FileModule.NAMESPACE_URI);
+    public static final QName FileUpdateElement = new QName("update", FileModule.NAMESPACE_URI);
+    public static final QName FileDeleteElement = new QName("delete", FileModule.NAMESPACE_URI);
+    public static final QName FileErrorElement = new QName("error", FileModule.NAMESPACE_URI);
+
+    // TODO(JL) Figure out which namespace all attributes should be in (possible breaking change)
+    public static final QName FileCollectionAttribute = new QName("collection", FileModule.NAMESPACE_URI);
+    public static final QName FileDirAttribute = new QName("dir", FileModule.NAMESPACE_URI);
+
+    public static final QName FileAttribute = new QName("file", XMLConstants.NULL_NS_URI);
+    public static final QName NameAttribute = new QName("name", XMLConstants.NULL_NS_URI);
+    public static final QName CollectionAttribute = new QName("collection", XMLConstants.NULL_NS_URI);
+    public static final QName TypeAttribute = new QName("type", XMLConstants.NULL_NS_URI);
+    public static final QName ModifiedAttribute = new QName("modified", XMLConstants.NULL_NS_URI);
+
+    public static final FunctionSignature signature =
```

Comment:
You could use the `FunctionDSL` here to simplify this

Reply:
This is not in scope of this PR